### PR TITLE
Support for ipfs-cluster pins

### PR DESCRIPTION
**Pull request by hsanjuan**
This makes pin-bot use ipfs-cluster to pin/unpin/check status (!status) from things.

It uses the legacy shells to resolve Cids and includes !legacypin and !legacyunpin.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -4,29 +4,36 @@ import (
...
-	hb "github.com/whyrusleeping/pinbot/Godeps/_workspace/src/github.com/whyrusleeping/hellabot"
+	shell "github.com/ipfs/go-ipfs-api"
+	hb "github.com/whyrusleeping/hellabot"
 )
 
 var prefix string
 var gateway string
 
 var (
-	cmdBotsnack = prefix + "botsnack"
```
Comment (hsanjuan) [inline main.go:20]: note that prefix here is empty, seemed like a left-over from the time a custom prefix flag was added.

**Comment by hsanjuan** [comment]
I'm pretty much done with this

**Comment by victorb** [comment]
My own comment from IRC

> the "so far pinned in 9/10 nodes" is a bit confusing, as this refers to actual ipfs-cluster nodes rather than nodes being pinned ("ipfs pin add --progress" gives you the number of nodes processed so far)

**Comment by victorb** [comment]
Text seems a bit weird here as well:

> 19:15:55 <pinbot> QmTF5diYwZFck3eHsy11VYGM6MqW1Y6Y1P5rowj9iyzzdv: reached pinned in all 10 required cluster peers. https://ipfs.io/ipfs/QmTF5diYwZFck3eHsy11VYGM6MqW1Y6Y1P5rowj9iyzzdv

**Comment by hsanjuan** [comment]
fixed those

**Pull request merged**